handlejob.py

```diff
@@ -0,0 +1,158 @@
+import os
+import sys
+import time
+from kubernetes import client, watch, config
+from kubernetes.client.rest import ApiException
+from subprocess import run
+
+from kubernetes.stream import stream
+
+
+def main():
+    job_name = os.environ["JOB_NAME"]
+    namespace = os.environ["NAMESPACE"]
+    timeout_minute_start_container = int(os.environ["TIMEOUT_MINUTE_START_CONTAINER"])
+
+    print(f"inputs are: \n jobname: {job_name} \n namespace: {namespace} \n timeout: {timeout_minute_start_container}")
+
+    v1, batch_api = init_k8s_configs()
+    job_uid = get_job_uid(batch_api, namespace, job_name)
+
+    timeout = time.time() + 60 * timeout_minute_start_container
+    print("Keep trying to get logs until backoffLimit has been reached (or Job succeed)")
+    while True:
+        print("Wait for the most recently created Pod to not be 'Pending' so logs can be fetched without errors")
+        pod = None
+        pod_is_ready = False
+        # wait until pod switched from 'pending' to one of 'Running' 'Failed' 'Succeeded' states
+        while not pod_is_ready and time.time() < timeout:
+            pod = get_pod_by_controller_uid(v1, namespace, job_uid)
+            if get_pod_phase(pod) != "Pending":
+                print("pod is ready")
+                pod_is_ready = True
+            time.sleep(10)
+        if not pod_is_ready:
+            print("timed out to start pod")
+            yell_and_exit_1(namespace, job_name)
+
+        print("""
+        Job is either 'Running' 'Failed' 'Succeeded'"
+        Attempt to fetch logs
+        -----------------------------
+        """)
+
+        tail_pod_log(v1, pod.metadata.name, namespace, job_name)
+
+        print("Job replica is done. checking Job status")
+        terminate_status = get_pod_terminate_status(v1, pod.metadata.name, namespace,  job_name)
+        pod_name = pod.metadata.name
+
+        if terminate_status == "Completed":
+            print("Job is successfully completed, fetching intermediate artifacts")
+            copy_output_from_extractor(namespace, pod_name)
+            trigger_extractor_container_termination(v1, namespace, pod_name)
+            sys.exit(0)
+        elif terminate_status == "Error":
+            # check if we reached limit of failed attempts and now just give up:
+            if backoff_limit_is_reached(batch_api, namespace, job_name):
+                print("Job has reach its backoffLimit and its final state is not 'complete', it ended with failures")
+                trigger_extractor_container_termination(v1, namespace, pod_name)
+                yell_and_exit_1(namespace, job_name)
+            # job is failed nut we still can try one more time, so stopping extractor
+            # to bring pod to finalised state, and it will get restarted by job controller.
```

Comment:
```suggestion
            # job failed, but we still can try one more time. Stop extractor to
            # bring pod to finalized state and it will get restarted by job controller.
```